lib/srv/db/postgres/users.go

```diff
@@ -210,9 +211,12 @@ func (e *Engine) applyPermissions(ctx context.Context, sessionCtx *common.Sessio
 	counts, countMap := permissions.CountObjectKinds(objsFetched)
 	e.Log.WithField("kind_counts", countMap).WithField("total", len(objsFetched)).Infof("Fetched %v objects from the database (%v).", len(objsFetched), counts)
 
-	objsTagged := permissions.ApplyDatabaseObjectImportRules(rules, sessionCtx.Database, objsFetched)
+	objsTagged, errs := databaseobjectimportrule.ApplyDatabaseObjectImportRules(rules, sessionCtx.Database, objsFetched)
 	counts, countMap = permissions.CountObjectKinds(objsTagged)
-	e.Log.WithField("kind_counts", countMap).WithField("total", len(objsFetched)).Infof("Tagged %v database objects (%v).", len(objsTagged), counts)
+	e.Log.WithField("kind_counts", countMap).WithField("total", len(objsFetched)).Infof("Tagged %v database objects (%v), errors: %v.", len(objsTagged), counts, len(errs))
+	for objName, err := range errs {
+		e.Log.WithField("name", objName).WithError(err).Debug("failed to apply label due to template error")
+	}
```

Comment:
should we just stop and return the error? I think we are usually confident that Apply should succeed. What cases it might fail? maybe newer auth with new templates in import rule?

Reply:
We don't have very tight control on the templates here, I imagine there may be some mechanisms with which invalid templates sneak through especially if we add more functionality in the future, like functions. This is why I don't think we should just return error and abort the login process; we can settle for localised failure here, i.e. lack of permissions for a specific object.